### DISCUSSION TITLE
Hide password using {log:false} in session based login

### DISCUSSION THREAD

**Discussion by DilawarQA23** [cypress-io/cypress#30415] [Questions and Help]
hi there,
I am new to Cypress.
I am using custom command to maintain the login of the user for further test cases.
I want to hide the password using  {log:false} in custom command of cypress but it does not working as expected.
the password (11221122) still displayed on cypress logs.

**My test.cy file code**

beforeEach(()=>
  {
     cy.login('test1@gmail.com','11221122')
  });

**my custom command code**

Cypress.Commands.add('login', (email, password) =>
     { 

        cy.session([email,password], ()=>
     {

 
      cy.visit('/')
     
      cy.get('#user_login').type(email)
    
      cy.get('#user_password').type(password,{log:false})
 
         })
     })

**Comment by portyu9** [comment]
Hi @DilawarQA23 

`{ log: false }` is working for the `.type()` command. The password is being exposed somewhere else: it is part of your **session ID** here:

```js
cy.session([email, password], ...)
```

Cypress displays the `cy.session()` ID in the reporter/Command Log, so sensitive values such as passwords or tokens should not be used as session identifiers.

Use a non-secret ID instead:

```js
Cypress.Commands.add('login', (email, password) => {
  cy.session(['login', email], () => {
    cy.visit('/')

    cy.get('#user_login').type(email)
    cy.get('#user_password').type(password, { log: false })

    // submit login...
  })
})
```

If two users can share the same email but require different session state, use another **non-sensitive** discriminator in the ID rather than the password.

On current Cypress, secrets should also be read with `cy.env()` rather than hard-coded in the spec.

Please mark this answer as accepted if it helped, thank you.
